packages/@aws-cdk/aws-stepfunctions-tasks/lib/sqs/send-message.ts

```diff
@@ -50,6 +57,140 @@ export interface SqsSendMessageProps extends sfn.TaskStateBaseProps {
+   *
+   * @param path the JSON path expression where the attributes are located.
+   *
+   * @returns a new `SqsMessageAttributes` instance.
+   *
+   * @see https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_MessageAttributeValue.html
+   */
+  public static fromJsonPathAt(path: string): SqsMessageAttributes {
+    return new SqsMessageAttributes(sfn.TaskInput.fromJsonPathAt(path));
+  }
+
+  /**
+   * Message attributes provided by an object literal.
+   *
+   * @param obj the message attributes to use.
+   *
+   * @returns a new `SqsMessageAttributes` instance.
+   */
+  public static fromObject(obj: { [key: string]: SqsMessageAttributeValue }): SqsMessageAttributes {
+    const rendered: { [key: string]: any; } = {};
+    for (const [key, value] of Object.entries(obj)) {
+      rendered[key] = value._taskInput.value;
+    }
+
+    return new SqsMessageAttributes(sfn.TaskInput.fromObject(rendered));
+  }
+
+  /** @internal */
+  public readonly _taskInput: sfn.TaskInput;
+
+  private constructor(value: sfn.TaskInput) {
+    this._taskInput = value;
+  }
+}
+
+/**
+ * Message attribute values.
+ */
+export class SqsMessageAttributeValue {
+  /**
+   * Attribute value provided directly from the task input. The value must be an
+   * object matching the `SQS.MessageAttributeValue` schema.
+   *
+   * @param path the JSON path expression where the attribute is located.
+   *
+   * @returns a new `SqsMessageAttributeValue` instance.
+   *
+   * @see https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_MessageAttributeValue.html
+   */
+  public static fromJsonPathAt(path: string): SqsMessageAttributeValue {
+    return new SqsMessageAttributeValue(sfn.TaskInput.fromJsonPathAt(path));
+  }
+
+  /**
+   * Binary attribute provided as a base64-encoded string.
+   *
+   * @param base64 the base64-encoded attribute value.
+   *
+   * @returns a new `SqsMessageAttributeValue` instance.
+   */
+  public static fromBase64(base64: string): SqsMessageAttributeValue {
+    return new SqsMessageAttributeValue(sfn.TaskInput.fromObject({ DataType: 'Binary', BinaryValue: base64 }));
+  }
+
+  /**
+   * Binary attribute provided from the task input.
+   *
+   * @param path the JSON expression where the base64-encoded attribute value is.
+   *
+   * @returns a new `SqsMessageAttributeValue` instance.
+   */
+  public static base64FromJsonPathAt(path: string): SqsMessageAttributeValue {
+    return new SqsMessageAttributeValue(sfn.TaskInput.fromObject({ DataType: 'Binary', BinaryValue: sfn.TaskInput.fromJsonPathAt(path).value }));
+  }
+
+  /**
+   * Number attribute.
+   *
+   * @param number the number value.
+   *
+   * @returns a new `SqsMessageAttributeValue` instance.
+   */
+  public static fromNumber(number: number): SqsMessageAttributeValue {
+    return new SqsMessageAttributeValue(sfn.TaskInput.fromObject({ DataType: 'Number', StringValue: `${number}` }));
```

Comment:
tokens confuse me still, but I'm pretty sure that the string interpolation will fail for tokens

Reply:
You are right, it might well result in unexpected behavior when the `number` is an unresolved token.